Tests/ComparisonTest.cs

```diff
@@ -160,30 +160,49 @@ public static string EvaluatorPrepareEnvironmentCode(Dictionary<string, string>
 
             JSFilenames = null;
 
+            AssemblyAppDomain = AppDomain.CreateDomain("TestAssemblyDomain", null, new AppDomainSetup
+            {
+                ApplicationBase = AppDomain.CurrentDomain.BaseDirectory,
+            });
+
             switch (extensions[0]) {
                 case ".exe":
                 case ".dll":
                     var fns = absoluteFilenames.ToArray();
                     if (fns.Length > 1)
                         throw new InvalidOperationException("Multiple binary assemblies provided.");
-
-                    Assembly = extensions[0] == ".exe" ? Assembly.ReflectionOnlyLoadFrom(fns[0]) : Assembly.LoadFrom(fns[0]);
+                    AssemblyUtility = CrossDomainHelper.CreateFromAssemblyPathOnRemoteDomain(AssemblyAppDomain, fns[0], extensions[0] == ".exe").AssemblyUtility;
```

Comment:
If we pass here AppDomain.CurrentDomain, it will work in same domain. We could introduce some options later to make it run faster (but with more memory usage/state effects).